features/FEATURE_COMMON_PAL/mbed-trace/source/mbed_trace.c

```diff
@@ -304,13 +304,7 @@ static void mbed_trace_default_print(const char *str)
 {
     puts(str);
 }
-void mbed_tracef(uint8_t dlevel, const char *grp, const char *fmt, ...)
-{
-    va_list ap;
-    va_start(ap, fmt);
-    mbed_vtracef(dlevel, grp, fmt, ap);
-    va_end(ap);
-}
+
```

Comment:
For proper flexibility your core routines in the new code must be `v` functions taking va_args, not variadic ones. So then you would be intercepting `mbed_vtracef`.